src/main/java/util/ApiQuotaManager.java

```diff
@@ -0,0 +1,181 @@
+
+    private static final Logger logger = HTLog.get(ApiQuotaManager.class);
+
+    /**
+     * The amount that is set aside for manual operations by the user.
+     */
+    private static final int API_QUOTA_BUFFER = 200;
+
+    /**
+     * The default refresh period.
+     */
+    private static final int DEFAULT_REFRESH_PERIOD_IN_MINS = 1;
+
+    /**
+     * The uninitialized value for uninitialized variables.
+     */
+    private static final int UNINITIALIZED = -1;
+
+    /**
+     * The amount of the available remaining API requests.
+     */
+    private int remainingApiRequests = UNINITIALIZED;
+
+    /**
+     * The number of API calls used in most recent refresh.
+     */
+    private int apiCallsUsedInPreviousRefresh = 0;
+
+    /**
+     * The duration between each refresh of the data store.
+     */
+    private long refreshDurationInMinutes = 1;
+
+    private final TickingTimer refreshTimer;
+
+    public ApiQuotaManager(Logic logic) {
+        this.refreshTimer = new TickingTimer("Refresh Timer",
+                (int) Utility.minsToSecs(DEFAULT_REFRESH_PERIOD_IN_MINS),
+                UI.status::updateTimeToRefresh, logic::refresh, TimeUnit.SECONDS);
+        this.refreshTimer.start();
+        UI.events.registerEvent((RefreshTimerTriggeredEventHandler) this::updateSyncRefreshRate);
+    }
+
+    /**
+     * @return The duration between each refresh of the data store.
+     */
+    public long getRefreshDurationInMinutes() {
+        return refreshDurationInMinutes;
+    }
+
+    /**
+     * Restarts the refreshTimer.
+     */
+    public void restartRefreshTimer(){
+        this.refreshTimer.restart();
+    }
+
+    /**
+     * Updates the period of the refreshTimer for synchronization of the data store.
+     * The initial call to this method does not update the period of the refreshTimer.
+     * @param e RefreshTimerTriggeredEvent object that holds the current API quota information.
+     */
+    private void updateSyncRefreshRate(RefreshTimerTriggeredEvent e) {
+        ApiQuotaInfo info = e.getApiQuotaInfo();
+
+        if (remainingApiRequests == UNINITIALIZED){
+            remainingApiRequests = info.getRemainingRequests();
+            return;
+        }
+
+        apiCallsUsedInPreviousRefresh = computeApiCallsUsedInPreviousRefresh(info.getRemainingRequests());
+        remainingApiRequests = info.getRemainingRequests();
+        refreshDurationInMinutes = computeRefreshTimerPeriod(info.getRemainingRequests(),
+                                                             Utility.minutesFromNow(info.getNextRefreshInMillisecs()),
+                                                             apiCallsUsedInPreviousRefresh, API_QUOTA_BUFFER,
+                                                             DEFAULT_REFRESH_PERIOD_IN_MINS);
+        this.refreshTimer.restartTimer((int) Utility.minsToSecs(refreshDurationInMinutes));
+        logger.info("Refresh period updated to " + refreshDurationInMinutes
```

Comment:
Add a blank line above this because end-of-method log message is a separate step.